src/results.cpp

```diff
@@ -945,6 +947,30 @@ REALM_RESULTS_TYPE(util::Optional<double>)
 
 #undef REALM_RESULTS_TYPE
 
+Results Results::freeze(SharedRealm frozen_realm) {
+    switch (m_mode) {
+        case Mode::Empty:
+            return Results();
+        case Mode::Table:
+            return Results(frozen_realm, *frozen_realm->transaction().import_copy_of(m_table));
+        case Mode::List: {
+            std::shared_ptr<LstBase> shared_list = std::move(frozen_realm->transaction().import_copy_of(*m_list));
+            return Results(frozen_realm, shared_list, m_descriptor_ordering);
+        }
+        case Mode::LinkList: {
+            std::shared_ptr<LstBase> shared_list = std::move(frozen_realm->transaction().import_copy_of(*m_link_list));
+            return Results(frozen_realm, shared_list); // FIXME: Do we need to parse on query and sort descriptor?
```

Comment:
This does need to do things for sort and distinct as those are implemented as a layer on top of the LnkLst.